dvc/repo/ls.py

```diff
@@ -21,20 +30,39 @@ def ls(url, target=None, rev=None, recursive=None, outs_only=False):
         if target and not result:
             raise PathMissingError(target, repo, output_only=outs_only)
 
-        def prettify(path_info):
-            if path_info == target_path_info:
-                return path_info.name
-            return relpath(path_info, target_path_info)
+        @colorize(with_color)
+        def prettify(fs_node):
+            if fs_node.path_info == target_path_info:
+                return fs_node.path_info.name
+            return relpath(fs_node.path_info, target_path_info)
 
-        result = list(set(map(prettify, result)))
-        result.sort()
+        result = {n.path_info: n for n in result}.values()
+        result = sorted(result, key=lambda n: n.path_info)
+        result = list(map(prettify, result))
     return result
 
 
+def colorize(with_color):
+    from dvc.utils.ls_colors import LsColors
+
+    ls_colors = LsColors(load=True) if with_color else None
+
+    def decorator(f):
+        @wraps(f)
+        def wrapper(fs_node):
+            if not ls_colors:
+                return f(fs_node)
+            return ls_colors.format(fs_node, f(fs_node))
+
+        return wrapper
+
+    return decorator
+
```

Comment:
It is quite weird that we have color knowledge in the API part. How about we return list of structures (FSNode i guess) that know if they are outs of some dvc file, or if they are dirs and then will colorize it as we wish in CLI part (dvc/command/ls.py)?